utils/http.hh

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/shared_future.hh>
+#include <seastar/coroutine/all.hh>
+#include <seastar/http/client.hh>
+#include <seastar/net/dns.hh>
+#include <seastar/net/tls.hh>
+
+#include "seastarx.hh"
+#include "log.hh"
+
+namespace utils::http {
+
+class dns_connection_factory : public seastar::http::experimental::connection_factory {
+protected:
+    std::string _host;
+    int _port;
+    logging::logger& _logger;
+    struct state {
+        bool initialized = false;
+        socket_address addr;
+        ::shared_ptr<tls::certificate_credentials> creds;
+    };
+    lw_shared_ptr<state> _state;
+    shared_future<> _done;
+
+    future<> initialize(bool use_https) {
```

Comment:
I think the word "TLS" should be used here (and below), not "HTTPS", and "TCP" instead of "HTTP". There's no real HTTP-specific code in this class...
Although maybe in a header file called http.hh, the word can remain, especially in log messages... Hmm...